### PR TITLE
Add TLS Cipher header

### PR DESCRIPTION
**Pull request by jadolg**
Ticket: OP-2637

## Description

This plugin takes TLS information and stores it in the request headers.
This PR sets the base to do so and takes care of setting the cipher used by the connection.

## How has this been tested

1. Execute the headers reader web service `make start_headers_reader`
2. Execute traefik with the plugin locally loaded `make testcontainer`
3. Make a request to traefik `curl https://localhost -k`
The response should contain a non-empty `X-Tls-Cipher` header.

### CODE REVIEW

File: .assets/icon.png
Comment (applejag) [inline .assets/icon.png:1]: Did you draw this? :O

I really like it!

File: go.sum
Comment (applejag) [inline go.sum:1]: Could potentially delete this empty file

File: plugin.go
Comment (applejag) [inline plugin.go:1]: Suggest doing some changes to the README.md.

What I think is most important is having some example of valid config, and what kind of output you get.

Maybe good idea to also mention that the cipher names should be "standard IDs" as they come from the https://pkg.go.dev/crypto/tls#CipherSuiteName function

**Review by applejag — CHANGES_REQUESTED** [review]

File: .assets/icon.png
Comment (jadolg) [reply .assets/icon.png:1]: fast google search :smile: I have 0 talent for this.

File: go.sum
Comment (jadolg) [reply go.sum:1]: I rather have an empty sum file for consistency so I don't forget about it when/if it gets data

File: .assets/icon.png
Comment (applejag) [reply .assets/icon.png:1]: Do we need to leave any attribution for it?

File: testconfig/dynamic.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      loadBalancer:
+        servers:
+          - url: "http://localhost:8888"
+
+  middlewares:
+    my-middleware:
+      plugin:
+        tlsheaders:
+          headers:
+            cipher: X-TLS-Cipher
```
Comment (applejag) [inline testconfig/dynamic.yml]: nit: canonicalize the header name

```suggestion
            cipher: X-Tls-Cipher
```

File: readme.md [hunk 2]
```diff
@@ -1,270 +1,98 @@
...
-
-Once loaded, middleware plugins behave exactly like statically compiled middlewares.
-Their instantiation and behavior are driven by the dynamic configuration.
+```yaml
+middlewares:
+  my-middleware:
+    plugin:
+      tlsheaders:
+        headers:
+          cipher: X-TLS-Cipher
```
Comment (applejag) [inline readme.md]: nit: canonicalize the header name

```suggestion
          cipher: X-Tls-Cipher
```

File: readme.md [hunk 3]
```diff
@@ -1,270 +1,98 @@
...
-            Foo: Bar
+apiVersion: traefik.io/v1alpha1
+kind: Middleware
+metadata:
+  name: my-middleware
+spec:
+  plugin:
+    tlsheaders:
+      headers:
+        cipher: X-TLS-Cipher
```
Comment (applejag) [inline readme.md]: nit: canonicalize the header name

```suggestion
        cipher: X-Tls-Cipher
```

File: plugin_test.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	if err == nil {
+		t.Fatal("expected error")
+	}
+}
+
+func TestTLSCipher(t *testing.T) {
+	cfg := CreateConfig()
+	cfg.Headers.Cipher = "X-TLS-Cipher"
+	next := http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+		assertHeader(t, r.Header, "X-TLS-Cipher", "TLS_AES_128_GCM_SHA256")
```
Comment (applejag) [inline plugin_test.go]: nit: canonicalize the header name

```suggestion
	cfg.Headers.Cipher = "X-Tls-Cipher"
	next := http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
		assertHeader(t, r.Header, "X-Tls-Cipher", "TLS_AES_128_GCM_SHA256")
```

File: .traefik.yml [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
 
-summary: '[Demo] Add Request Header'
+summary: 'Add TLS information to request headers'
 
 testData:
-  Headers:
-    X-Demo: test
-    X-URL: '{{URL}}'
+  headers:
+    cipher: X-TLS-Cipher
```
Comment (applejag) [inline .traefik.yml]: nit: canonicalize the header name

```suggestion
    cipher: X-Tls-Cipher
```

**Review by applejag — CHANGES_REQUESTED** [review]
nit: canonicalize the header name

File: .assets/icon.png
Comment (jadolg) [reply .assets/icon.png:1]: done

**Review by applejag — APPROVED** [review]
LGTM!

**Pull request merged**